src/liballoc/allocator.rs

```diff
@@ -464,6 +464,30 @@ impl fmt::Display for CannotReallocInPlace {
 ///  * if a layout `k` fits a memory block (denoted by `ptr`)
 ///    currently allocated via an allocator `a`, then it is legal to
 ///    use that layout to deallocate it, i.e. `a.dealloc(ptr, k);`.
+///
+/// # Unsafety
+///
+/// The `Alloc` trait is an `unsafe` trait for a number of reasons, and
+/// implementors must ensure that they adhere to these contracts:
+///
+/// * Pointers returned from allocation functions must point to valid memory and
+///   retain their validity until at least the instance of `Alloc` is dropped
+///   itself.
+///
+/// * Implementations of `Alloc` intended to be used for global allocators are
+///   not currently memory safe if they unwind.  This restriction may be lifted
```

Comment:
I would prefer "it's undefined behavior if a global allocator unwinds" to "not currently memory safe".

Reply:
"not memory safe" implies there's an edge case in which UB can happen, but allocators unwinding is *always* UB.